model/riscv_pmp_regs.sail

```diff
@@ -154,15 +163,34 @@ function pmpReadCfgReg(n) = {
 }
 
 /* Helpers to handle locked entries */
+function pmpLockBit(cfg: Pmpcfg_ent) -> bool =
+  cfg.L() == 0b1
+   
 function pmpLocked(cfg: Pmpcfg_ent) -> bool =
-   cfg.L() == 0b1
+  (cfg.L() == 0b1) & (mseccfg.RLB() == 0b0)
 
 function pmpTORLocked(cfg: Pmpcfg_ent) -> bool =
-   (cfg.L() == 0b1) & (pmpAddrMatchType_of_bits(cfg.A()) == TOR)
-
+  (cfg.L() == 0b1) & (mseccfg.RLB() == 0b0) & (pmpAddrMatchType_of_bits(cfg.A()) == TOR)
+  
+/* Function to check whether executable region can be added or not in accordance to 4b of Smepmp spec */
+function checkExe(cfg: Pmpcfg_ent,v: bits(8)) -> Pmpcfg_ent =
```

Comment:
checkExe is a poor name for this

Reply:
And I don't understand why this isn't just inlined in pmpWriteCfg, it's just one extra check

Reply:
Yes right... removed the function